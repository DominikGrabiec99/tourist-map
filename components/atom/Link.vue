<template>
  <a v-if="isGlobalLink" :href="href" :_blank="blank">
    <slot />
  </a>
  <NuxtLink v-else :to="href">
    <slot />
  </NuxtLink>
</template>

<script lang="ts" setup>
const props = withDefaults(defineProps<{
  href: string,
  blank?: boolean,
}>(), {
  href: '',
  blank:  false,
})

const { href } = toRefs(props);

const isGlobalLink = computed(() => href.value.includes('http'));
</script>