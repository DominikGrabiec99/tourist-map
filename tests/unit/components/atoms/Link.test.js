import { describe, it, expect, beforeAll } from 'vitest';
import Link from '@/components/atom/Link.vue';
import { mount } from '@vue/test-utils';
describe('Icon.vue', () => {
  let wrapper;

  beforeAll(() => {
    wrapper = mount(Link, {
      props: {
        href: 'https://www.test.pl',
        blank: true,
      },
    });
  });

  it('should render "a" html element', () => {
    expect(wrapper.vm.isGlobalLink).toBeTruthy();
    const link = wrapper.get('[data-test="html-link"]');
    expect(link.element.getAttribute('_blank')).toBeTruthy();
    expect(link).toBeDefined();
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render nuxtLink', () => {
    const wrapper = mount(Link, {
      props: {
        href: 'test-link',
      },
    });
    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.get('[data-test="nuxt-link"]')).toBeDefined();
  });
});
