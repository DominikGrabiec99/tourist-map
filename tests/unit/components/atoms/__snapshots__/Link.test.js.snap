// Vitest Snapshot v1

exports[`Icon.vue > render component 1`] = `
<a
  _blank="true"
  data-test="html-link"
  href="https://www.test.pl"
>
  
  
</a>
`;

exports[`Icon.vue > should render "a" html element 1`] = `
<a
  _blank="true"
  data-test="html-link"
  href="https://www.test.pl"
>
  
  
</a>
`;

exports[`Icon.vue > should render nuxtLink 1`] = `
<nuxtlink
  data-test="nuxt-link"
  to="test-link"
>
  
  
</nuxtlink>
`;
